### ISSUE TITLE
about the usage of "detach()"

### ISSUE THREAD

**Issue by liujianzhao6328057** [zwx8981/UNIQUE#15]
Hi, sorry to have interrupted again. I wonder why we need to add detach() when computing the probability and the hingeloss.   

       p = 0.5 * (1 + torch.erf(y_diff / torch.sqrt(2 * y_var.detach())))
                std_label = torch.sign((gstd1 - gstd2))                self.std_loss = self.std_loss_fn(y1_var, y2_var, std_label.detach())

**Comment by zwx8981** [comment]
@liujianzhao6328057 In the hingloss，std_label is the ground-truth, which should be detached during training in PyTorch. When computing the probability, it is better (empirically) to disentangle the learning of quality prediction and uncertainty quantification modules for favorable optimization behaviors.
